tests/e2e/provisioning_e2e/tests/test_async_symmetric_enrollments.py

```diff
@@ -103,29 +106,27 @@ def assert_device_provisioned(device_id, registration_result):
     :param device_id: The device id
     :param registration_result: The registration result
     """
-    assert registration_result.status == "assigned"
-    assert registration_result.registration_state.device_id == device_id
-    assert registration_result.registration_state.assigned_hub == linked_iot_hub
+    print(registration_result)
+    assert registration_result["status"] == "assigned"
+    assert registration_result["registrationState"]["deviceId"] == device_id
+    assert registration_result["registrationState"]["assignedHub"] == linked_iot_hub
 
```

Comment:
Why are we changing `registration_result`, etc, from objects to dictionaries?  This seems like a step backwards.

Reply:
that is because of this. we don't have translation from what the service gives us to our special object..if we want to keep things as it is then we ca define RegistrationResult on our own...and translate things like "deviceId" to "device_id"